src/appshell/view/preferences/appearancepreferencesmodel.h

```diff
@@ -78,12 +81,17 @@ class AppearancePreferencesModel : public QObject, public async::Asyncable
     QColor foregroundColor() const;
     QString foregroundWallpaperPath() const;
 
+    Q_INVOKABLE void resetThemeToDefault();
+    Q_INVOKABLE bool enableHighContrastChecked();
+    Q_INVOKABLE void loadLastUsedGeneralTheme();
```

Comment:
Could you please explain the purpose of this method (same for loadLastUsedHighContrastTheme)?

Reply:
These methods are called upon checking/unchecking the "Enable High-Contrast" checkbox.
Whenever i switch between the modes (General/High-Contrast) (by checking/unchecking the checkbox), the active theme would be whatever theme i last used in that category. I think this adds a little touch of convenience when switching between themes, both for the user who may have customized a theme in both categories perhaps, and for the developer who may be testing features/differences between a general and HC theme.

for example,
1. I have the Dark theme active
2. I then check the high-contrast checkbox, and i will get defaulted to the HC-Black theme (only when switching to HC-Mode for the *first* time)
3. I switch to the HC-White theme instead, because it looks cool
4. I uncheck the checkbox, and am taken back to whichever general theme I used last, which in this case was Dark theme
5. Upon checking the box again, I would have the HC-White theme active again, since that is the theme i used last in the HC Category

The last used General/HC Theme values are stored in `UI_PREVIOUS_GENERAL_THEME` and `UI_PREVIOUS_HIGH_CONTRAST_THEME` variables respectively, in `uiconfiguration.cpp`.